R/Utils.R

```diff
@@ -451,3 +451,48 @@ scale_rows <- function(x){
   s <- apply(x, 1, sd, na.rm = T)
   return((x - m) / s)
 }
+
+PrependDatasetIdToClonotypeBarcodes <- function(clonotype_directory, output_directory){
+  
+  #initialize the output directory
+  if (!dir.exists(output_directory)) {
+    dir.create(output_directory, recursive = T)
+  }
+  #populate the list of clonotype files to iterate on
+  files <- list.files(clonotype_directory, full.names = T)
+  
+  #iterate on clonotype files, prepending barcodePrefix/DatasetId to the barcodes
+  for (file in files){
+    print(paste("Parsing", file))
+    #isolate & store final directory in path (file name)
+    file_name <- tail(unlist(strsplit(file, split = "/")),1)
+    #grab the first numeric string (datasetId)
+    datasetId <- strsplit(file_name, split = "\\.")[[1]][1]
+    clonotype_csv_df <- read.csv(file) |> as.data.frame()
+    clonotype_csv_df$barcode <- paste0(datasetId, "_", clonotype_csv_df$barcode)
+    clonotype_csv_df$barcode <- gsub("-1", "", clonotype_csv_df$barcode)
+    #when writing the concatenated CSV, keep the colnames of the first file, else discard them. 
+    output_file_name <- file_name
+    
+    write.table(clonotype_csv_df, 
+                file = paste0(output_directory,output_file_name), 
+                append = F, 
+                sep = ",", 
+                row.names = F, 
+                col.names = T)
+  }
+  
+}
+
+SeuratToCoNGA <- function(seuratObj, TCR_indir, TCR_outdir, GEX_dir) {
+  seuratObj <- Rdiscvr::DownloadAndAppendTcrClonotypes(seuratObj, TCR_indir)
```

Comment:
The test "fails" because we don't import Rdiscvr, but I think the plan is to make a change in Rdiscvr so that it has a bespoke `PrependDatasetIdToClonotypeBarcodes()` function and writes a merged clonotypes csv (e.g. CellBarcode = DatasetId+CellBarcode). 

However, I think a `SeuratToConga()` function takes the seuratObj, performs `write10xCounts()` and then kicks off the conga entry point in python (which will need some assurances on where the TCR data downloaded from that to-be-implemented Rdiscvr file is) is desirable. 

I'll add the RDiscvr import now, with the intention of us keeping this PR and changing it to import the bespoke TCR function.

Reply:
for posterity: Rdiscvr is prime-seq aware, while CellMembrane is not (by design). So anything that needs to query prime-seq (i.e. TCR data) is more in line over there. 

https://github.com/bimberlabinternal/Rdiscvr/pull/27 covers what this PR was seeking to do.